### PR TITLE
f-cookiebanner@v0.6.0 - Add component and A11y tests

### PR DESCRIPTION
**Pull request by JoshuaNg2332**
v0.6.0
------------------------------
*March 18, 2021*

### Added
- New tests to test f-cookiebanner in `component`
- New test supporting files in `test-utils/compoenent-objects`
- Browserstack support and file structures in tests
- A11y tests

### Fixed
- URL for NZ cookie policy

### CODE REVIEW

File: packages/components/organisms/f-cookie-banner/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@justeat/f-cookie-banner",
   "description": "Fozzie Cookie Banner – Cookie Banner",
-  "version": "0.5.0",
+  "version": "0.5.1",
```
Comment (simonsandersje) [inline packages/components/organisms/f-cookie-banner/package.json]: I think this PR warrants more than a minor version bump - suggest 0.6.0

File: packages/components/organisms/f-cookie-banner/src/components/CookieBanner.vue [hunk 2]
```diff
@@ -8,15 +8,16 @@
...
         data-cookie-consent-overlay
-        data-test-id="cookieBanner-component"
         :aria-hidden="shouldHideBanner">
         <div
             :class="[
                 $style['c-cookieBanner-card'],
                 { [$style['c-cookieBanner-ios']]: isIosBrowser }
-            ]">
+            ]"
+            data-test-id="newCookieBanner-component">
```
Comment (simonsandersje) [inline packages/components/organisms/f-cookie-banner/src/components/CookieBanner.vue]: Not a good habit using "new" IMHO - it's only new for the duration of this PR! I have been using "cookieConsentBanner" to show the difference with the legacy banners

File: packages/components/organisms/f-cookie-banner/test-utils/component-objects/f-cookieBanner-new.component.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
+const Page = require('@justeat/f-wdio-utils/src/page.object');
+
+module.exports = class CookieBanner extends Page {
+
+    get component () { return $('[data-test-id="newCookieBanner-component"]')}
```
Comment (simonsandersje) [inline packages/components/organisms/f-cookie-banner/test-utils/component-objects/f-cookieBanner-new.component.js]: See above

File: packages/components/organisms/f-cookie-banner/test/specs/accessibility/axe-accessibility.spec.js [hunk 4]
```diff
@@ -1,14 +1,27 @@
-const CookieBanner = require('../../../test-utils/component-objects/f-cookieBanner.component');
-const cookieBanner = new CookieBanner();
 import { getAccessibilityTestResults } from '../../../../../../../test/utils/axe-helper';
+const LegacyCookieBanner = require('../../../test-utils/component-objects/f-cookieBanner-legacy.component');
+const legacyCookieBanner = new LegacyCookieBanner();
+const NewCookieBanner = require('../../../test-utils/component-objects/f-cookieBanner-new.component');
```
Comment (simonsandersje) [inline packages/components/organisms/f-cookie-banner/test/specs/accessibility/axe-accessibility.spec.js]: See above

File: packages/components/organisms/f-cookie-banner/test-utils/component-objects/f-cookieBanner-legacy.component.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    isCookieBannerComponentDisplayed() {
+        return this.component.isDisplayed();
+    }
+
+    clickCookiePolicyLink() {
+        this.cookiePolicyLink.click();
+    }
+
+    close() {
```
Comment (LTurns) [inline packages/components/organisms/f-cookie-banner/test-utils/component-objects/f-cookieBanner-legacy.component.js:25]: qq - do you need to `return this` here?

File: packages/components/organisms/f-cookie-banner/test/specs/component/desktop/f-cookieBanner-new.component.spec.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    });
+
+    forEach([['full'], ['necessary']])
+    .it('should set "je-cookie_banner" and "je-cookieConsent" to expected cookie values', expectedCookieValue => {
+        // Act
+        cookieBanner.acceptCookies(expectedCookieValue);
+
+        const bannerCookie = browser.getCookies().filter(cookie => cookie.name === 'je-banner_cookie')[0];
+        const bannerConsent = browser.getCookies().filter(cookie => cookie.name === 'je-cookieConsent')[0];
+
```
Comment (LTurns) [inline packages/components/organisms/f-cookie-banner/test/specs/component/desktop/f-cookieBanner-new.component.spec.js]: just missing some `asserts` and `arranges` here I think?

File: packages/components/organisms/f-cookie-banner/test/specs/component/shared/f-cookieBanner-legacy.component.spec.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+const CookieBanner = require('../../../../test-utils/component-objects/f-cookieBanner-legacy.component');
+const cookieBanner = new CookieBanner();
+import forEach from 'mocha-each';
+
+describe('Legacy - Multi-tenant - f-cookieBanner component tests', () => {
+    forEach([['gb'], ['au'], ['nz']])
```
Comment (LTurns) [inline packages/components/organisms/f-cookie-banner/test/specs/component/shared/f-cookieBanner-legacy.component.spec.js]: when it's only one item, I don't think we need to add sub-arrays - it should work with just `['gb', 'au', 'nz'] ` 😄

File: packages/components/organisms/f-cookie-banner/test/specs/component/shared/f-cookieBanner-new.component.spec.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    });
+
+    it('should display the f-cookieBanner accept necessary button', () => {
+        // Assert
+        expect(cookieBanner.isCookieBannerAcceptNecessaryButtonDisplayed()).toBe(true);
+    });
+});
+
+describe('New - Multi-tenant - f-cookieBanner component tests', () => {
+    forEach([['dk'], ['es'], ['ie'], ['it'], ['no']])
```
Comment (LTurns) [inline packages/components/organisms/f-cookie-banner/test/specs/component/shared/f-cookieBanner-new.component.spec.js]: see above 😄

File: packages/components/organisms/f-cookie-banner/test-utils/component-objects/f-cookieBanner-legacy.component.js (hunk 5 above)
Comment (JoshuaNg2332) [reply packages/components/organisms/f-cookie-banner/test-utils/component-objects/f-cookieBanner-legacy.component.js:25]: We don't need it for here as we're just doing a click function on this.

File: packages/components/organisms/f-cookie-banner/test/specs/component/shared/f-cookieBanner-legacy.component.spec.js (hunk 7 above)
Comment (siggerzz) [reply packages/components/organisms/f-cookie-banner/test/specs/component/shared/f-cookieBanner-legacy.component.spec.js]: We tried this and I don't _think_ it works, mocha-each will think it's a single test with three params!

File: packages/components/organisms/f-cookie-banner/test/specs/component/shared/f-cookieBanner-new.component.spec.js (hunk 8 above)
Comment (siggerzz) [reply packages/components/organisms/f-cookie-banner/test/specs/component/shared/f-cookieBanner-new.component.spec.js]: ditto

**Review by simonsandersje — APPROVED** [review]

**Review by LTurns — APPROVED** [review]

**Pull request merged**